documentation/specs/src/interoperability/ethereum-bridge/bootstrapping.md

```diff
@@ -1,117 +1,115 @@
-
-The active validator set at the launch height chosen for starting the Ethereum
-bridge will have the extra responsibility of restarting the chain if they
-consider the deployed smart contracts as valid. For this reason, the validator
-set at this height must be known in advance of the governance proposal
-resolving, and a channel set up for offchain communication and co-ordination of
-the chain restart. In practise, this means the governance proposal to launch the
-chain should commit to doing so within an epoch of passing, so that the
-validator set is definitely known in advance.
-
-### Deployer
-
-Once the smart contracts are fully deployed, only the active validator set for
-block height `h` should have control of the contracts so in theory anyone could
-do the Ethereum bridge smart contract deployment.
-
-### Backing out of Ethereum bridge launch
+## Overview
 
-If for some reason the validity of the smart contract deployment cannot be
-agreed upon by the validators who will responsible for restarting Namada, it
-must remain possible to restart the chain with the Ethereum bridge still not
-enabled.
+To bootstrap the Ethereum bridge, there are six governance parameters which
+must be written to storage:
+
+- `eth_bridge_min_confirmations` - The minimum number of block confirmations
+  on Ethereum required for any given event to be voted on by Namada validators.
+- `eth_bridge_bridge_address` - The address of the `Bridge` contract, used to
+  perform transfers in either direction (Namada <> Ethereum).
+- `eth_bridge_bridge_version` - The version of the `Bridge` contract, starting
+  from 1.
+- `eth_bridge_governance_address` - The address of the `Governance` contract,
+  used to perform administrative tasks, such as updating validator sets in
+  Ethereum.
+- `eth_bridge_governance_version` - The version of the `Governance` contract,
+  starting from 1.
+- `eth_bridge_wnam_address` - The address of the deployment of the native
+  ERC20 token, representing NAM in Ethereum.
+
+Here is an overview of the steps to follow to enable the Ethereum bridge through a
+governance proposal for a given Namada chain:
+
+1. The [smart contracts](./ethereum_smart_contracts.md) of the Ethereum bridge
+   should be deployed a priori, with an initial validator set and other parameters,
+   such as nonces, from epoch $E_0$. This step can be done by anyone, but ultimately
+   it's up to governance validators to decide which addresses to trust.
+2. A governance proposal should be held to agree on an epoch $E_{bridge} > E_0$ at
+   which to launch the Ethereum bridge. The proposal should be executed at some
+   grace epoch $E_{grace}$, with $E_{grace} \le E_{bridge}$. The wasm code of
+   the proposal should modify storage with:
+
+    - The epoch $E_{bridge}$ when the bridge is to be enabled.
+    - The aforementioned governance parameters of the Ethereum bridge.
+
+   Additionally, the knowledge of $E_0$ must be communicated through the governance
```

Comment:
I assume a hash of the validator set will be used?

Reply:
the epoch should be enough, since you can query the validator set of a given epoch